lib/media/drm_engine.js

```diff
@@ -1012,8 +1032,14 @@ shaka.media.DrmEngine = class {
    */
   createTemporarySession_(initDataType, initData) {
     let session;
+
+    const mediaKeySystemAccessConfiguration =
+      this.mediaKeySystemAccess_.getConfiguration();
```

Comment:
If the only thing you're getting is `persistentStateRequired`, why not get that from the `DrmInfo`?  Also note the `persistentState` value will almost always be `required` because at this point, the only other valid value is `not-allowed` and that doesn't happen much.  Also note that some platforms don't return a configuration object at all.

Reply:
I've done it that way because of what i've experienced when testing with Edge
https://github.com/google/shaka-player/issues/1495
> Edge is able to read content when PlayReady Server SDK enforce MediaLicense Type to NonPersistent ONLY if persistentState is set to not-allowed. Passing optional to persistentState results in the MediaKeySystemAccess Configuration to have persistentState to required

But I'll implement it as you stated